src/runs.rs

```diff
@@ -0,0 +1,341 @@
+use chrono::{DateTime, Local};
+#[cfg(unix)]
+use daemonize::Daemonize;
+#[cfg(unix)]
+use miette::IntoDiagnostic;
+use names::Generator;
+use serde::{Deserialize, Serialize};
+use std::{fmt, path::PathBuf};
+use sysinfo::{Pid, ProcessStatus, System};
+
+use crate::{consts, Project};
+
+#[derive(Debug)]
+pub struct DaemonRunsManager<'a> {
+    pub project: &'a Project,
+}
+
+impl<'a> DaemonRunsManager<'a> {
+    pub fn new(project: &'a Project) -> Self {
+        let daemon_runs = Self { project };
+
+        // Create the runs directory if it doesn't exist
+        std::fs::create_dir_all(daemon_runs.runs_dir()).expect("Failed to create runs directory");
+
+        daemon_runs
+    }
+
+    pub fn runs_dir(&self) -> PathBuf {
+        self.project.pixi_dir().join(consts::RUNS_DIR)
+    }
+
+    pub fn runs(&self) -> Vec<DaemonRun> {
+        // NOTE(hadim): the source of truth for managed runs are the pid files (any files ending with `.pid`) in the runs directory.
+
+        let runs: Vec<DaemonRun> = std::fs::read_dir(self.runs_dir())
+            .expect("Failed to read runs directory")
+            .filter_map(|entry| {
+                let entry = entry.expect("Failed to read entry");
+                let path = entry.path();
+                let file_name = path.file_name().expect("Failed to get file name");
+                let file_name = file_name
+                    .to_str()
+                    .expect("Failed to convert file name to str");
+                if file_name.ends_with(".pid") {
+                    let run_name = file_name.replace(".pid", "");
+
+                    Some(DaemonRun::new(
+                        run_name,
+                        self.runs_dir(),
+                        self.project.root().to_path_buf(),
+                    ))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        runs
+    }
+
+    pub fn create_new_run(&self, name: Option<String>) -> miette::Result<DaemonRun> {
+        // Check if a run with the same name already exists
+        let name = match name {
+            Some(name) => name,
+            None => {
+                // We generate a random name for this run
+                let mut generator = Generator::default();
+                generator.next().expect("Failed to generate random name")
+            }
+        };
+
+        // Check not the same name as an existing run
+        if self.runs().iter().any(|run| run.name == name) {
+            miette::bail!("A run with the same name already exists. You can call `pixi runs clear` to clear all the terminated runs.");
+        }
+
+        Ok(DaemonRun::new(
+            name,
+            self.runs_dir(),
+            self.project.root().to_path_buf(),
+        ))
+    }
+
+    pub fn get_run(&self, name: String) -> miette::Result<DaemonRun> {
```

Comment:
Would it make sense to just return a `Option` here?

Reply:
Can you explain why? Handling the error here is convenient since in the cli commands such as clear or kill I can just use `let run = runs_manager.get_run(args.name)?;` and automatically handle the error for me.

So it makes the consumers of that function less verbose.

Let me know what you think.

Reply:
Closing here but please re-open if needed.